src/run/linker.rs

```diff
@@ -0,0 +1,341 @@
+use super::*;
+
+/// Stores extra data needed about the nodes when in lazy mode. (In strict mode,
+/// this is unused.)
+pub(super) struct Header {
+  /// the principal port of this node
+  pub(super) this: Port,
+  /// the port connected to the principal port of this node
+  pub(super) targ: Port,
+}
+
+pub struct Linker<'h, M: Mode> {
+  pub(super) allocator: Allocator<'h>,
+  pub redexes: Vec<(Port, Port)>,
+  pub rwts: Rewrites,
+  headers: IntMap<Addr, Header>,
+  _mode: PhantomData<M>,
+}
```

Comment:
This needs a doc comment.
```suggestion
/// [Example: The state of a linker, which links together wires and ports while resolving thread conflicts.]
///
/// This struct does [...]. It is necessary because [...].
pub struct Linker<'h, M: Mode> {
  pub(super) allocator: Allocator<'h>,
  pub redexes: Vec<(Port, Port)>,
  pub rwts: Rewrites,
  headers: IntMap<Addr, Header>,
  _mode: PhantomData<M>,
}
```